editor.planx.uk/src/@planx/components/FileUploadAndLabel/SelectMultiple.tsx

```diff
@@ -40,194 +45,176 @@ const ListHeader = styled(Box)(({ theme }) => ({
-      );
-      setFileList(updatedFileList);
-    }
-
-    if (removedTags && removedTags.length > 0) {
-      const updatedFileList = removeSlots(removedTags, uploadedFile, fileList);
-      setFileList(updatedFileList);
+    switch (reason) {
+      case "selectOption": {
+        const updatedTags = selectedTags.filter(
+          (tag) => !initialTags?.includes(tag),
+        );
+        const updatedFileList = addOrAppendSlots(
+          updatedTags,
+          uploadedFile,
+          fileList,
+        );
+        setFileList(updatedFileList);
+        break;
+      }
+      case "removeOption": {
+        const removedTags = initialTags?.filter(
+          (tag) => !selectedTags?.includes(tag),
+        );
+        const updatedFileList = removeSlots(
+          removedTags,
+          uploadedFile,
+          fileList,
+        );
+        setFileList(updatedFileList);
+        break;
+      }
     }
   };
 
-  useEffect(() => {
-    updateFileListWithTags(previousTags, tags);
-  }, [tags]);
+  /**
+   * Options for autocomplete
+   * FileList with appended "category" property for grouping
+   */
+  const options: Option[] = (
+    Object.keys(fileList) as Array<keyof typeof fileList>
+  )
+    .filter((fileListCategory) => fileList[fileListCategory].length > 0)
+    .flatMap((category) =>
+      fileList[category].map((fileType) => ({ category, ...fileType })),
+    );
+
+  /**
+   * Previous values to pre-populate autocomplete
+   */
+  const value: Option[] = initialTags.flatMap((tag) =>
+    options.filter(({ name }) => name === tag),
+  );
+
+  /**
+   * Custom listbox component
+   * Used to wrap options within the autocomplete and append a custom element above the option list
+   */
+  const ListboxComponent = forwardRef<typeof Box, PropsWithChildren>(
+    ({ children, ...props }, ref) => {
+      return (
+        <Box ref={ref} {...props} role="listbox">
+          <ListHeader>
+            <ListSubheader disableGutters>
+              <Typography variant="h4" component="h3" pr={3}>
+                Select all that apply
+              </Typography>
+              <Button
+                variant="contained"
+                color="prompt"
+                onClick={closePopper}
+                aria-label="Close list"
+              >
+                Done
+              </Button>
+            </ListSubheader>
+          </ListHeader>
+          {children}
+        </Box>
+      );
+    },
+  );
 
   return (
     <FormControl
       key={`form-${uploadedFile.id}`}
       sx={{ display: "flex", flexDirection: "column" }}
     >
-      <StyledInputLabel
-        id={`select-multiple-file-tags-label-${uploadedFile.id}`}
-      >
-        What does this file show?
-      </StyledInputLabel>
-      <StyledSelect
-        native={false}
-        key={`select-${uploadedFile.id}`}
-        id={`select-multiple-file-tags-${uploadedFile.id}`}
-        variant="standard"
-        multiple
-        value={tags}
+      <Autocomplete
```

Comment:
I know this is still WIP - but just thought I'd flag prior MUI `<Autocomplete />` a11y PRs for prop reference!
- https://github.com/theopensystemslab/planx-new/pull/865 - this one highlights some obscure ones like `aria-atomic`, `handleHomeEndKeys` etc that it doesn't look like we're supporting here yet
 - https://github.com/theopensystemslab/planx-new/pull/887 - really detailed notes about expected behavior per device in this one (hopefully MUI component has improved enough in last ~2 years that we'll be able to pass using it this time!)

Reply:
This is super helpful thanks - will take a detailed look through.